### DISCUSSION TITLE
Overwrite default ipynb entrypoint for my_st-nb (like `nb_custom_formats` only for ipynb)

### DISCUSSION THREAD

**Discussion by kai-tub** [executablebooks/meta#613] [Q&A] [answered]
I've originally posted the _issue_ in [executablebooks/jupyter-book](https://github.com/executablebooks/jupyter-book/issues/1586) [issue1586](https://github.com/executablebooks/jupyter-book/issues/1586).
Now I understand that the issue is _caused_ by the MyST-NB extension.
Or by the way sphinx registers these function.
I have a custom preprocessor function that I would _like_ to apply to all ipynb Notebooks, before running them through jupyter-book.
It works very well, but only if I register it via a different extension (`.my_ipynb` for example).
By looking at the [myst-nb documentation for custom nb formats](https://myst-nb.readthedocs.io/en/latest/examples/custom-formats.html#custom-notebook-formats), I would expect that the following setting would work (as it is written in the documentation)
```python
nb_custom_formats = {
    '.ipynb': ['nbformat.reads', {'as_version': 4}],
}
```
But this triggers sphinx to re-register the extension and raises an error.
```python
    def add_nb_custom_formats(app: Sphinx, config):
    """Add custom conversion formats."""
    for suffix in config.nb_custom_formats:
        app.add_source_suffix(suffix, "myst-nb")
```
from [here](https://github.com/executablebooks/MyST-NB/blob/2cc1260b175b1bafe6f80f903e942569930b1f84/myst_nb/__init__.py#L359)

I am not very familiar with sphinx and mainly use jupyter-book.
Is there a simple fix for this _problem_?
I understand that it technically is not a bug in jupyter-book or myst-nb, as the name contains the word _custom_ but due to the documentation, it sounds like it should be possible to modify the entry-point.

Any help would be greatly appreciated. :)

**Comment by kai-tub** [comment]
I would think that by simply checking if `suffix` is one of `ipynb` or `md` it would be possible to not re-register it and cause sphinx to trigger an error.
I am not so sure, if it would fix the underlying issue though

**Comment by chrisjsewell** [comment, marked as answer]
Heya, sorry you beat me to a reply lol, but I've literally just added this in https://github.com/executablebooks/MyST-NB/blob/3e54270c2252ebfd119ea7b80460765fe5693d0d/myst_nb/sphinx_.py#L119, plus the requisite logic for choosing the format reader in https://github.com/executablebooks/MyST-NB/blob/3e54270c2252ebfd119ea7b80460765fe5693d0d/myst_nb/read.py#L61

This is a big refactor of myst-nb, but nearly there, so hopefully in jupyter-book in a reasonable time

**Reply by kai-tub** [reply]
I do not know how I always end up finding these _corner-cases_ and end up installing directly from GitHub :D 
Thanks for pointing me to it!

I am having trouble finding the branch where the development is happening from the permalink.
Do you mind telling me which branch I should be subscribing to?
I need to judge if I should document my current workaround for my colleagues or if I should just wait.
I do _not_ expect any release estimate, I am more than happy to check the git history on my own :D

**Reply by chrisjsewell** [reply]
Its `refactor-mystnb`, but I would maybe wait a day or to, until I open a draft PR and explain it, since it pretty much re-writes the whole package lol (in a mainly back-compatible manner)
